qiskit/_register.py

```diff
@@ -19,19 +19,21 @@
 Base register reference object.
 """
 import re
+import logging
+
 from ._qiskiterror import QISKitError
 
+logger = logging.getLogger(__name__)
+
 
 class Register(object):
     """Implement a generic register."""
 
     def __init__(self, name, size):
         """Create a new generic register."""
-        test = re.compile('[a-z][a-zA-Z0-9_]*')
-        if test.match(name) is None:
-            raise QISKitError("invalid OPENQASM register name")
         self.name = name
         self.size = size
+        self.oq_name = None
```

Comment:
because there is no need if the openqasm representation is never used (and we are trying to go in that direction).

I changed `oq_name` for `_openqasm_name` (to avoid collision the method)

Reply:
`openqasm_name()` might be a property

Reply:
done in e5a5fe4